File: src/main/java/superstitio/orbs/actions/AnimationOrbOnMonsterAction.java
package superstitio.orbs.actions;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.core.Settings;
import superstitio.orbs.orbgroup.OrbGroup;

public class AnimationOrbOnMonsterAction extends AbstractGameAction {
    public OrbGroup target;

    public AnimationOrbOnMonsterAction(final OrbGroup target, final int amt) {
        this.target = target;
        this.amount = amt;
        this.duration = Settings.ACTION_DUR_FAST;
    }

    public void update() {
        if (this.duration != Settings.ACTION_DUR_FAST) return;
        for (int i = 0; i < this.amount; i++) {
            target.triggerEvokeAnimation(i);
        }
        this.isDone = true;
    }
}

File: superstitioMain/java/superstitio/orbs/actions/AnimationOrbOnMonsterAction.kt
package superstitio.orbs.actions

import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.core.Settings
import superstitio.orbs.orbgroup.OrbGroup

class AnimationOrbOnMonsterAction(target: OrbGroup, amt: Int) : AbstractGameAction() {
    var targetOrbGroup: OrbGroup = target

    init {
        this.amount = amt
        this.duration = Settings.ACTION_DUR_FAST
    }

    override fun update() {
        if (this.duration != Settings.ACTION_DUR_FAST) return
        for (i in 0 until this.amount) {
            targetOrbGroup.triggerEvokeAnimation(i)
        }
        this.isDone = true
    }
}
